core/src/main/java/com/google/instrumentation/stats/TagKey.java

```diff
@@ -30,27 +35,14 @@
    * </ol>
    */
   public static TagKey create(String key) {
-    return new TagKey(key);
-  }
-
-  @Override
-  public boolean equals(Object obj) {
-    return (obj instanceof TagKey) && key.equals(((TagKey) obj).key);
+    return new AutoValue_TagKey(StringUtil.sanitize(key));
   }
 
-  @Override
-  public int hashCode() {
-    return key.hashCode();
-  }
+  public abstract String asString();
 
+  // TODO(sebright) Use the default AutoValue toString(), which is more useful for debugging.
```

Comment:
gRPC is currently calling toString() to access the key and value fields.  I added that to the comment.